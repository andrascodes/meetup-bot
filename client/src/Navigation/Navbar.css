.NavDropdownIcon {
  font-size: 1.35em;
}

.NavDropdownUsername {
  font-size: 1.2em;
  font-weight: bold;
}

@media only screen and (max-width : 768px) {
  .navbar-nav>li {
      float: left;
  }

  .navbar-right {
    float: right!important;
    margin-right: -15px;
  }
}
