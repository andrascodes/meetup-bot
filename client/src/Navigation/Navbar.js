import React from 'react'

import { Link, withRouter } from 'react-router-dom'
import { NavDropdown, MenuItem } from 'react-bootstrap'

import { createAuthApi } from '../lib/authApi'

import './Navbar.css'

export const Navbar = withRouter(props => {

  const logOut = createAuthApi(fetch).logout

  const handleSignOutClick = () => {
    logOut()
    .then(res => {
      localStorage.clear()
      props.history.push('/login')
    })
    .catch(error => {
      console.log(error)
      localStorage.clear()
      props.history.push('/login')
    })
  }

  const handleProfileClick = () => {
    props.history.push('/profile')
  }

  return (
    <nav className="navbar navbar-inverse navbar-fixed-top">
      <div className="container-fluid">
        <div className="navbar-header">
          <Link to="/">
            <div className="navbar-brand">{props.brand}</div>
          </Link>
        </div>

        <ul className="nav navbar-nav navbar-right">
          <NavDropdown 
            title={<i className="fa fa-user NavDropdownIcon" />}
            id="NavDropdownMenu"
          >
            <MenuItem header >
              <div className="NavDropdownUsername">
                Signed in as {`'${localStorage.username}'`}
              </div>
            </MenuItem>
            <MenuItem divider/>
            <MenuItem onClick={handleProfileClick}>Profile</MenuItem>
            <MenuItem onClick={handleSignOutClick}>Sign out</MenuItem>
          </NavDropdown>
        </ul>
      </div>
    </nav>
  )
})
