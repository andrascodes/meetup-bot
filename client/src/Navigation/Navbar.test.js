import React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter, Router } from 'react-router-dom'
import createHistory from 'history/createBrowserHistory'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'

import { Navbar } from './Navbar'

describe('Navbar', () => {

  beforeEach(() => {
    global.localStorage = {
      username: 'admin',
      clear: jest.fn()
    }
  })

  test('renders correctly', () => {
    let tree = renderer.create(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('redirects to /profile when clicking the Profile link', () => {
    const history = createHistory()
    const wrapper = mount(
      <Router history={history}>
        <Navbar/>
      </Router>
    )
    const locationBefore = history.location.pathname
    const expectedLocationAfter = '/profile'
    wrapper.find('.ProfileLink').find('a').simulate('click')
    const actualLocationAfter = history.location.pathname
    expect(actualLocationAfter).not.toEqual(locationBefore)
    expect(actualLocationAfter).toEqual('/profile')
  })

  test('handles SignOut link click', () => {
    const mockedHandleSignoutClick = jest.fn()
    const history = createHistory()
    const wrapper = mount(
      <Router history={history}>
        <Navbar onSignOutClick={mockedHandleSignoutClick}/>
      </Router>
    )
    wrapper.find('.SignOutLink').find('a').simulate('click')
    expect(mockedHandleSignoutClick).toHaveBeenCalled()
  })

})