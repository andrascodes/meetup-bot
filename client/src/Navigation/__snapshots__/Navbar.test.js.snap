exports[`Navbar renders correctly 1`] = `
<nav
  className="navbar navbar-inverse navbar-fixed-top">
  <div
    className="container-fluid">
    <div
      className="navbar-header">
      <a
        href="/"
        onClick={[Function]}>
        <div
          className="navbar-brand" />
      </a>
    </div>
    <ul
      className="nav navbar-nav navbar-right">
      <li
        className="dropdown"
        style={undefined}>
        <a
          aria-expanded={false}
          aria-haspopup={true}
          className="dropdown-toggle"
          href="#"
          id="NavDropdownMenu"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button">
          <i
            className="fa fa-user NavDropdownIcon" />
           
          <span
            className="caret" />
        </a>
        <ul
          aria-labelledby="NavDropdownMenu"
          className="dropdown-menu"
          role="menu">
          <li
            className="dropdown-header"
            onKeyDown={[Function]}
            role="heading"
            style={undefined}>
            <div
              className="NavDropdownUsername">
              Signed in as 
              \'admin\'
            </div>
          </li>
          <li
            className="divider"
            onKeyDown={[Function]}
            role="separator"
            style={undefined} />
          <li
            className="ProfileLink"
            role="presentation"
            style={undefined}>
            <a
              href="#"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1">
              Profile
            </a>
          </li>
          <li
            className="SignOutLink"
            role="presentation"
            style={undefined}>
            <a
              href="#"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1">
              Sign out
            </a>
          </li>
        </ul>
      </li>
    </ul>
  </div>
</nav>
`;
